### ISSUE TITLE
Flash of incorrect theme in jQuery mode

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#1125]
Similar to, but not exactly the same issue as #787, I notice this flashing particularly in jQuery mode. When in light theme, it flashes to dark between article loads, and when in dark theme, it flashes to light. This is not the case in SW mode. The fact that the flash is the opposite of the loaded theme implies it is not a hard-coded value, and should be fixable.

**Comment by infinityanant** [comment]
I would like to work on this issue! I’ve analyzed the problem and my plan is to move the theme detection logic into a blocking script in the <head> to prevent the Flash of Unstyled Content (FOUC) in jQuery mode. Could you please assign this to me?

**Comment by infinityanant** [comment]
I recorded a short video demonstrating the issue:
https://drive.google.com/file/d/12iMRFjNHeS0dQ4Co-kpbrf20pFbrYRnu/view?usp=sharing

In the recording, the site is initially in dark theme. I then switch the network to 3G and refresh the page. The page first loads in light mode and then switches back to dark mode.

@Jaifroid Is this the same problem you were referring to earlier?

**Comment by Jaifroid** [comment]
@infinityanant I think you might be addressing #787 rather than this issue, which is specifically about JQuery mode? It's possible your PR could fix both - needs testing.

**Comment by infinityanant** [comment]
@Jaifroid You're right while investigating the jQuery mode issue, I realized the white flash was happening at the document level in index.html before any mode was even initialized.
By moving the theme-check to the <head>, it seems to resolve the global 'white flash' (#787) while also ensuring that jQuery mode (#1125) doesn't start with a white canvas.
I am happy to make any necessary adjustments based on your feedback or the test results. Please let me know if there are specific changes you'd like to see!
